pkg/server/service.go

```diff
@@ -42,10 +51,30 @@ type CRIContainerdService interface {
 }
 
 // criContainerdService implements CRIContainerdService.
-type criContainerdService struct{}
+type criContainerdService struct {
+	// rootDir is the directory for managing cri-containerd files.
+	rootDir string
+	// containerService is containerd container service client.
+	containerService execution.ContainerServiceClient
+	// sandboxStore stores all sandbox metadata.
+	sandboxStore metadata.SandboxStore
+	// sandboxNameIndex stores all sandbox names and make sure each name
+	// is unique.
+	sandboxNameIndex *registrar.Registrar
+	// sandboxIDIndex is trie tree for partial id index.
+	sandboxIDIndex *truncindex.TruncIndex
+}
```

Comment:
For image, NameIndex is not needed, and store may not be needed in the future, either.